frontend/packages/kubevirt-plugin/src/components/modals/add-template-source/add-template-source.tsx

```diff
@@ -82,6 +96,7 @@ export const AddTemplateSourceModal: React.FC<ModalComponentProps &
   const { dataSource, file, isValid } = state;
 
   const onSubmit = async () => {
+    setError(undefined);
     setAllocating(true);
```

Comment:
IMO we should not `setAllocating(true)` until the certificate check is successful, because it causes the modal to flicker between checks.
Lets create a new boolean `isCheckingCert` and pass it to the `ModalFooter`  `inProgress ` prop